### ISSUE TITLE
torch.pinverse for complex tensors and general complex tensor support

### ISSUE THREAD

**Issue by AqibHasnain** [pytorch/pytorch#45385]
## 🚀 Feature
The SVD based torch.pinverse which works with complex tensors would be great. 

## Motivation

For low-rank matrices, this is an absolute must. And of course we can use the torch.svd functionality and then write the pseudoinverse ourselves, however this would make life a bit simpler. 

## Edit

One overarching more issue I've come across is the lack of functionality in PyTorch for complex tensors. Although I would like a pinverse function, I can still take the pseudoinverse of a complex tensor via the SVD. But then I ran into the issue that matrix multiplication of complex tensors is not supported. So I wrote a function to compute complex matmul. Then I went to start training the network and was hit with the warning that complex tensor gradients aren't fully supported. I am running PyTorch 1.6.0 by the way. 

So I guess my question would be if these issues are on the radar of the PyTorch team? And have you found that others would like this functionality as well? I, for one, would greatly benefit from full complex tensor support! Thanks for reading!

cc @ezyang @anjali411 @dylanbespalko @vincentqb @vishwakftw @jianyuh @nikitaved @pearu

**Comment by anjali411** [comment]
Hi @AqibHasnain thanks for the feature request! Complex Number Support is definitely on the radar and is being worked upon. you can track the progress for various ongoing tasks here: https://github.com/pytorch/pytorch/issues/33152.  From what I know, a lot of linear algebra functions (for complex input) surface in audio applications, so the community will certainly benefit from the extended support for complex numbers. As for complex autograd, pytorch uses tensorflow style gradients for complex numbers, which would ensure that you can solve optimization problems with real valued loss function and complex parameters. hope that answered some of your questions. If you would like to contribute to pick one of the complex number related issues or work to add `torch.pinverse`, let me know as well.

**Comment by AqibHasnain** [comment]
@anjali411 thanks! If I get a bit of time, I will certainly let you know about contributing to the complex numbers support. 

Thanks for the clarification on complex autograd. The warning that autograd gave had me worried that the complex parameters would be an issue.
